### ISSUE TITLE
Duration + no_box results in zero-length durations being serialized to empty fields in the protobuf message

### ISSUE THREAD

**Issue by poohsen** [scalapb/ScalaPB#1730]
Hi. 
I assume that this is a design decision: An empty field mapped to a Duration without boxing must be mapped onto _some_ value and that value is _zero_. Then, to remain symmetrical during serialization, a Duration of zero without a box is serialized to an empty field. 

Ideally I would like to serialize to a Duration of zero even if it's boxed. On the deserialization side, both an empty field and a 0 would map to a Duration of zero. Is that possible?

**Comment by poohsen** [comment]
> Trying to understand the issue and motivation: why the serialized representation matter to your project?

We expected the duration never to be empty and were surprised in a typescript client when it turned out to be empty for zero stations.

> Would a solution where you define your own custom Duration type in a proto2 file and typemap it to a ScalaPB Duration on the Scala side would be useful


Yes that would work. Though the actual duration value couldn't be serialized to a uint64 because that will serialized the zero to an empty field, too, IIUC. It would have to be a string representation. 

Though by now I've seen some protobuf docs that mentioned those default values and that one cannot know if e.g. a duration was empty or zero when deserializing. So this isn't scalapb-specific I guess.

**Comment by thesamet** [comment]
Since you mentioned typescript: are we taking about serialization to bytes
or to JSON?

**Comment by poohsen** [comment]
> are we taking about serialization to bytes
or to JSON?
I mean serialization to bytes. But I imagine that my questions come across as a bit confused. I don't have enough experience with protobuf to tell where protobuf ends and scalapb begins plus I'm only learning the ins and outs of protobuf as I go.

To put it in code, we have these messages:
```
message Time {google.protobuf.Duration quantity = 1 [(scalapb.field).no_box = true];}

message CType {
  oneof sealed_value {
    Unspecified unspecified = 1; 
    B b = 2;   
    Time time = 3;   
  }
}
```
and we observed that a Duration of `0` resulted in an empty field in the serialized protobuf message (we peeked inside using an online viewer). We then wrongly concluded that this was because of scalapb's `no_box = true` but have since understood that `com.google.protobuf.duration.Duration` with value `0` will always be empty in protobuf.

This makes me wonder why the typescript code that we generate doesn't turn an `undefined` Duration into one of zero length but that's another question altogether.

TL;DR: i was confused and this can be closed. Apologies for the noise.
